File: Dart/lib/shortestToChar.dart
import 'dart:math';

class Solution {
  List<int> shortestToChar(String s, String c) {
    var n = s.length;
    var res = List.filled(n, 0);
    var prev = -n;
    for (var i = 0; i < n; i++) {
      if (s[i] == c) {
        prev = i;
      }
      res[i] = i - prev;
    }
    prev = 2 * n;
    for (var i = n - 1; i >= 0; i--) {
      if (s[i] == c) {
        prev = i;
      }
      res[i] = min(res[i], prev - i);
    }
    return res;
  }
}

File: Typescript/src/shortestToChar.ts
export function shortestToChar(s: string, c: string): number[] {
	const n = s.length;
	const result: number[] = new Array(n).fill(Infinity);

	let prev = -Infinity;
	for (let i = 0; i < n; i++) {
		if (s[i] === c) {
			prev = i;
		}
		result[i] = Math.min(result[i], i - prev);
	}

	prev = Infinity;
	for (let i = n - 1; i >= 0; i--) {
		if (s[i] === c) {
			prev = i;
		}
		result[i] = Math.min(result[i], prev - i);
	}

	return result;
};
